### DISCUSSION TITLE
Pycharm unexpected type(s) for date field

### DISCUSSION THREAD

**Discussion by SergeyKapshuchenko** [fastapi/sqlmodel#848] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Hero(SQLModel, table=True):
    __tablename__ = "hero"
    created_date: date = Field(default_factory=lambda: datetime.now(UTC).date())


def get_min_created_date() -> date:
    with get_session() as session:
        query = select(Hero.created_date).where(Statistic.id == 1)
        created_date = session.exec(query).first()
```

### Description

Hi, pycharm complains about unexpected type when I select fields of date type.

1. Create model Hero
2. Create field created_date: date of date type
3. select it using select(Hero.created_date)
4. pycharm warning about - Unexpected type(s): (date) Possible type(s): (TypedColumnsClauseRole[_T0] | SQLCoreOperations[_T0] | Type[_T0]) (_TScalar_0) 
5. return type for value is also not working - created_date: Row | RowMapping | None

<img width="928" alt="Знімок екрана 2024-03-18 о 22 03 13" src="https://github.com/tiangolo/sqlmodel/assets/50737888/488ef7bc-6ae2-4fac-a986-2890a1c6a4fb">
<img width="928" alt="Знімок екрана 2024-03-18 о 22 03 16" src="https://github.com/tiangolo/sqlmodel/assets/50737888/eb0a8a1c-f066-4d62-983b-fb13d8a19987">

### Operating System

macOS

### SQLModel Version

0.0.16

### Python Version

3.12

**Comment by SergeyKapshuchenko** [comment]
but works great with datetime field

**Comment by SergeyKapshuchenko** [comment]
any updates ?

**Reply by anschweitzer** [reply]
Same issue with PyCharm 2025.1.1.1 and SQLModel 0.0.24, using [this tutorial code](https://sqlmodel.tiangolo.com/tutorial/where/#__tabbed_7_1)

**Comment by monchin** [comment]
From [the source code](https://github.com/fastapi/sqlmodel/blob/0.0.24/sqlmodel/sql/_expression_select_gen.py#L38-L51)
```python
_TScalar_0 = TypeVar(
    "_TScalar_0",
    Column,  # type: ignore
    Sequence,  # type: ignore
    Mapping,  # type: ignore
    UUID,
    datetime,
    float,
    int,
    bool,
    bytes,
    str,
    None,
)
```
You can see that `date` is not supported. Instead you should use `datetime` for better lint support. Of course I think your request is quite reasonable, so maybe you could make a PR to add `date` in `_TScalar_1/2/3` to fix this issue.

**Comment by YuriiMotov** [comment, marked as answer]
Use `col()`:

```py
        query = select(col(Hero.created_date))
```

---

<img width="590" height="159" alt="image" src="https://github.com/user-attachments/assets/92838223-e1e6-499d-aaeb-c9cc84742d62" />

---

<img width="590" height="159" alt="image" src="https://github.com/user-attachments/assets/22310597-c317-40d1-81cc-d96238e8c43f" />
